### ISSUE TITLE
Prohibit adding POIs if maps are not updated

### ISSUE THREAD

**Issue by gtrus21** [organicmaps/organicmaps#7930]
Please prohibit making changes to OSM if the maps of this region are not updated to the latest version.  Otherwise, edit conflicts, duplicates, etc. arise.

**Comment by Jean-BaptisteC** [comment]
That's already the case, if map is outdated.
Button to access to editor is not visible.

**Comment by patepelo** [comment]
> That's already the case, if map is outdated. Button to access to editor is not visible.

What user has updated a region but not the one he's editing?

**Comment by gtrus21** [comment]
Today one user created a POI, although I had already created this POI before him, and his map was not updated

**Comment by Jean-BaptisteC** [comment]
Can you share the position of POI?

**Comment by gtrus21** [comment]
It turned out that the edit of that user was never uploaded, because...  it was not registered.  But the point is that this POI was created on his phone, which means the editor allowed it to be created, despite the fact that the maps were old

**Comment by gtrus21** [comment]
The POI was created 24 days ago, I have it on my maps dated 03/26/24.  The user does not have one, so he did not update the cards.  The user is a complete newbie who only recently learned about OSM

**Comment by biodranik** [comment]
Let's clarify the issue: editing of POIs on the outdated map is DISABLED in OM (although a user can edit some features before updating the app and maps).

Adding new POIs is always enabled now, even if maps are outdated. So a warning should be displayed with a button to update the current map if a user selects "Add place". 
@Jean-BaptisteC correct?

**Comment by pastk** [comment]
When editing is disabled it should be also obvious to the user why its disabled. Now the button just doesn't show up and this is confusing.

**Comment by biodranik** [comment]
> So maybe we should control only the enabling, not the visibility?

Right, "Edit Place" should be visible only for existing objects (e.g. we don't show it for empty map places selected by a long tap).

The next question is how to make it obvious to users that the map needs to be updated. Note that for non-downloaded areas a download map button is displayed in the bottom bar under the Place Page. Maybe we should always display "Update maps" button there in such cases?

**Comment by kirylkaveryn** [comment]
> Maybe we should always display "Update maps" button there in such cases?

Or "Update map to edit". And by tapping on the button we may open the downloading screen.

**Comment by Kopterlove** [comment]
If I didn't  update the app, OM know what need to be updated?

User can be without internet for weeks or months. If he already made tons of changes, and make connect to internet and update the app... changes can be uploaded without issue?

**Comment by map-per** [comment]
I took a look at the **Android** behavior:

When the map is too old both the editing and adding buttons are removed from the place page.

User can, however, still create new features by using the add option from the menu (burger icon -> 'Add Place to OpenStreetMap')

**Comment by gtrus21** [comment]
What does "too old" mean?

**Comment by map-per** [comment]
Editing the app version is disabled when one of these two things happens:
- The Map data is more than 3 months older than the latest app data available for the currently running App version
- The map data is more than 6 months old

So if you regularly update the app you can edit for 3 months, but if you also don't update the app it's 6 months

---

The relevant code from [storage/storage.cpp](https://github.com/organicmaps/organicmaps/blob/32da9eda665aeebfe9502b3939bec606e065c7a0/storage/storage.cpp):

```
// Editing maps older than approximately three months old is disabled, since the data
// is most likely already fixed on OSM. Not limited to the latest one or two versions,
// because a user can forget to update maps after a new app version has been installed
// automatically in the background.
uint64_t const kMaxSecondsTillLastVersionUpdate = 3600 * 24 * 31 * 3;
// Editing maps older than approximately six months old is disabled, because the device
// may have been offline for a long time.
uint64_t const kMaxSecondsTillNoEdits = 3600 * 24 * 31 * 6;
```

```
return currentVersionTime - localVersionTime < kMaxSecondsTillLastVersionUpdate && 
             base::SecondsSinceEpoch() - localVersionTime < kMaxSecondsTillNoEdits;
```

**Comment by pastk** [comment]
- https://github.com/organicmaps/organicmaps/pull/8752
